cpp/dolfinx/common/Timer.cpp

```diff
@@ -7,21 +7,14 @@
 #include "Timer.h"
 #include "TimeLogManager.h"
 #include "TimeLogger.h"
+#include <optional>
 #include <stdexcept>
 
 using namespace dolfinx;
 using namespace dolfinx::common;
 
 //-----------------------------------------------------------------------------
-Timer::Timer() : Timer::Timer("")
-{
-  // Do nothing
-}
-//-----------------------------------------------------------------------------
-Timer::Timer(const std::string& task) : _task(task)
-{
-  // Do nothing
-}
+Timer::Timer(std::optional<std::string> task) : _task(std::move(task)) {}
```

Comment:
Just make a copy rather than move - no performance need for the extra complexity.